app/src/chainlit.py

```diff
@@ -47,16 +70,19 @@ def _init_chat_engine(engine_id: str) -> ChatEngineInterface | None:
     return None
 
 
-async def _init_chat_settings(engine: ChatEngineInterface) -> dict[str, Any]:
+def _init_chat_settings(
+    engine: ChatEngineInterface, query_values: dict[str, str]
+) -> list[InputWidget]:
     input_widgets: list[InputWidget] = [
-        _WIDGET_FACTORIES[setting_name](getattr(engine, setting_name))
+        _WIDGET_FACTORIES[setting_name](
+            query_values.pop(setting_name, None)
+            or getattr(app_config, setting_name, None)
+            or getattr(engine, setting_name)
```

Comment:
URL query parameters take precedence over `app_config` constants, which take precedence over engine default settings. This sets the initial value for the setting in the widget, which can subsequently be modified by the user.